### PR TITLE
Session-TimeOut 

### PR DESCRIPTION
**Pull request by bbare**
Implemented SessionTimeOut login in the ValidateAndUpdateSession method.

# Issue/Ticket

PROGRESS ON #13 

# Description

Logic to check if a current session is valid or not. If valid, return the updated session. If not, delete the current session.

## Type of change

List one of the following: Feature

Label your PR accordingly.

# How Has This Been Tested?

Tests to come. 

# Notes

### CODE REVIEW

File: Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs [hunk 1]
```diff
@@ -37,18 +37,20 @@ public Session CreateSession(DatabaseContext _db, User user)
...
-            {
-                return null;
-            }
-        }
-
-        public Session DeleteSession(DatabaseContext _db, string token)
+			//For the Compare method, a -1 is returned if the current Session expiration is shorter than the current time
+			//A 0 is returned if they are equal
+			//A 1 is returned if current Session is longer than the current time
+			if (TimeSpan.Compare(response.ExpiresAt.TimeOfDay, DateTime.UtcNow.TimeOfDay) == -1 || TimeSpan.Compare(response.ExpiresAt.TimeOfDay, DateTime.UtcNow.TimeOfDay) == 0)
```
Comment (alfredovargasjr) [inline Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs]: what about the case of someone starting the session at 11:50pm? and the check it done after midnight? just wondering

File: Backend/UnitTesting/UnitTesting.csproj [hunk 2]
```diff
@@ -60,11 +60,13 @@
     <Reference Include="System" />
     <Reference Include="System.Core" />
     <Reference Include="System.Data" />
+    <Reference Include="System.Data.Entity" />
```
Comment (alfredovargasjr) [inline Backend/UnitTesting/UnitTesting.csproj:63]: was this added to fix build errors?

File: Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs (hunk 1 above)
Comment (bbare) [reply Backend/ManagerLayer/AuthorizationManagement/AuthorizationManager.cs]: I will look into it. Im pretty sure the TimeSpan.Compare method will take care of it. Will update when I have an answer.

File: Backend/UnitTesting/UnitTesting.csproj (hunk 2 above)
Comment (bbare) [reply Backend/UnitTesting/UnitTesting.csproj:63]: Yes.

File: Backend/DataAccessLayer/Database/dbContext.cs [hunk 3]
```diff
@@ -14,12 +14,12 @@ namespace DataAccessLayer.Database
     public class DatabaseContext : DbContext
     {
 
-        public DatabaseContext()
```
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/Database/dbContext.cs]: Do not commit this

File: Backend/DataAccessLayer/Models/Session.cs [hunk 4]
```diff
@@ -14,6 +14,13 @@ public Session()
             ExpiresAt = DateTime.UtcNow.AddMinutes(MINUTES_UNTIL_EXPIRATION);
             Id = Guid.NewGuid();
         }
+		public Session(DateTime CreatedAt, DateTime UpdatedAt, DateTime ExpiresAt)
```
Comment (alfredovargasjr) [inline Backend/DataAccessLayer/Models/Session.cs:17]: dont think you need this overloaded constructor, you can just set the values via accessor

**Review by alfredovargasjr — CHANGES_REQUESTED** [review]

**Review by alfredovargasjr — DISMISSED** [review]

**Review by alfredovargasjr — APPROVED** [review]

**Pull request merged**